test/e2e/proxy_test.go

```diff
@@ -406,17 +404,6 @@ func TestNodePortAndEgressWithTheSameBackendPod(t *testing.T) {
 	skipIfProxyAllDisabled(t, data)
 	skipIfEncapModeIsNot(t, data, config.TrafficEncapModeEncap) // Egress works for encap mode only.
 
-	cc := func(config *controllerconfig.ControllerConfig) {
-		config.FeatureGates["Egress"] = true
-	}
-	ac := func(config *agentconfig.AgentConfig) {
-		config.FeatureGates["Egress"] = true
-	}
-
-	if err := data.mutateAntreaConfigMap(cc, ac, true, true); err != nil {
-		t.Fatalf("Failed to enable Egress feature: %v", err)
-	}
```

Comment:
even though we do not need to mutate the ConfigMap, I think we should still have a skip statement in the test case in case Egress has been disabled (like we do for Antrea policy e2e tests). We should probably do the same thing for egress_test.go test cases as well.

Reply:
I do not see we explicitly disable Egress in any tests.

The only case that is skipped by Egress test but this one is skipIfAntreaIPAMTest(t). I could add it to this test too.

Reply:
I meant that we should introduce `skipIfEgressDisabled`, like we have `skipIfAntreaPolicyDisabled`: https://github.com/antrea-io/antrea/blob/245ac54fbfbf5f0e9def6740612a1a8a127e3659/test/e2e/antreapolicy_test.go#L81

Reply:
But if we egress is enabled by default and we never disable it in tests, why we need that?

Reply:
Or are you asking me to create mode in which Egress is disabled (in the controller/agent config file)? My confusion is from as we never have such a mode, and the only to disable Egress at this moment is to edit ConfigMap and restart controller/agent, why I should add `skipIfEgressDisabled` to this test.

Reply:
I'll try to phrase it differently...
It's possible for someone to modify the config map today and deploy Antrea to a cluster with the Egress feature disabled.
Then it's possible to run Antrea e2e tests with
```
go test ./test/e2e/...
```
e.g., to validate a specific configuration
I am of the opinion that all test cases that require the Egress feature should be skipped in that case, so that the overall test run can succeed. We do this for AntreaPolicy, and I believe [AntreaProxy as well](https://github.com/antrea-io/antrea/blob/516f72e5b7c5d1e04e0bf1209166fdd7327b0b2b/test/e2e/fixtures.go#L156-L158), even though these 2 features are enabled by default.

Reply:
Oh, ok. I got what you mean now. Added skipIfEgressDisabled.